File: film.cpp
#include <string>

using namespace std;

class film
{
	private: 
		string id;
		string nama;
		string genre;
		int harga;
		int jual;
	public: 
		//constructor I 
		film(){
			id = "kosong";
		}

		//constructor II (ada paremeter)
		film(string id, string nama, string genre, int harga, int jual){
			this->id = id;
			this->nama = nama;
			this->genre = genre;
			this->harga = harga;
			this->jual = jual;
		}

		//destructor
		~film(){}

		//getter
		string getid(){
			return id;
		}
		string getnama(){
			return nama;
		}
		string getgenre(){
			return genre;
		}
		int getharga(){
			return harga;
		}
		int getjual(){
			return jual;
		}

		//setter
		void setid(string text){
			id = text;
		}
		void setnama(string text){
			nama = text;
		}
		void setgenre(string text){
			genre = text;
		}
		void setharga(int angka){
			harga = angka;
		}
		void setjual(int angka){
			jual = angka;
		}
};

File: PHP/film.php
<?php
class Film {
    private $id;
    private $nama;
    private $genre;
    private $harga;
    private $jual;

    // Constructor mimicking the C++ implementation
    public function __construct($id = "kosong", $nama = "", $genre = "", $harga = 0, $jual = 0) {
        $this->id = $id;
        $this->nama = $nama;
        $this->genre = $genre;
        $this->harga = $harga;
        $this->jual = $jual;
    }

    // Getters
    public function getId() { return $this->id; }
    public function getNama() { return $this->nama; }
    public function getGenre() { return $this->genre; }
    public function getHarga() { return $this->harga; }
    public function getJual() { return $this->jual; }

    // Setters
    public function setId($text) { $this->id = $text; }
    public function setNama($text) { $this->nama = $text; }
    public function setGenre($text) { $this->genre = $text; }
    public function setHarga($angka) { $this->harga = $angka; }
    public function setJual($angka) { $this->jual = $angka; }
}
?>